kernel/Base/MeshManipulator_Impl.h

```diff
@@ -1536,6 +1555,38 @@ void MeshManipulator<DIM>::readMesh(const std::string &filename) {
         "threads, please rerun the preprocessor first",
         numberOfPartitions, MPIContainer::Instance().getNumberOfProcessors());
 
+    // ZONES //
+    ///////////
+
+    std::vector<Zone *> zones;
+    if (version == 1) {
+        // Zones are only present in version > 1, provide a default zone
+        Zone *zone = addZone("Main");
```

Comment:
This api is a bit counterintuitive for me, we first add a zone and then push it back as well. I would somehow expect the addZone to push_back as well. I am not sure I make sense.

Reply:
It is just to have a simple list of pointers, so that I can later easily do `zones[index]` instead of `&(getZones()[index])` or something similar. I'll add a comment to it to clarify it.